bn-apps/cordapp-updates-distribution/corda-updates-shell/src/main/kotlin/net/corda/businessnetworks/cordaupdates/shell/CordaUpdatesCLI.kt

```diff
@@ -0,0 +1,166 @@
+ * folder by default unless a custom location has been provided via --configPath flag
+ * - SYNC. Pulls down missing CorDapps from configured remote repositories. Downloads all versions of all CorDapps from all remote repositories,
+ * that are configured in the settings.conf file. If a custom CorDapp / version range has been specified via --cordapp flag, the tool will attempt to
+ * find a respective cordapp source in the settings.conf and will fail if one doesn't exist.
+ * - PRINT_VERSIONS. Prints available versions of a specified CorDapp to the console. Will fail if no CorDapp name has been provided.
+ */
+class CordaUpdatesCLI : CordaCliWrapper("corda-updates", "CLI for corda-updates utility.") {
+    companion object {
+        val logger : Logger by lazy { LoggerFactory.getLogger("corda-updates") }
+        const val CONFIG_FILE_NAME = "settings.conf"
+    }
+
+    //TODO: this is a temporary hack until subcommands are supported: https://r3-cev.atlassian.net/browse/CORDA-1838
+    @CommandLine.Option(
+            names = ["--mode", "-m"],
+            required = true,
+            description = ["The execution mode. \${COMPLETION-CANDIDATES}."],
+            defaultValue = "SYNC",
+            showDefaultValue = CommandLine.Help.Visibility.ALWAYS)
+    private lateinit var command : CLICommands
+
+    @CommandLine.Option(
+            names = ["--cordapp", "-c"],
+            description = ["Coordinates of a CorDapp in standard Maven notation. Supports version ranges i.e. [0,10.1).",
+                "If not specified then all CorDapps specified in the $CONFIG_FILE_NAME will be synced."])
+    private var cordapp : String? = null
+
+    @CommandLine.Option(
+            names = ["--configPath", "-cp"],
+            description = ["Path to the $CONFIG_FILE_NAME file. If not specified, the file will be looked for in the current folder and ~/.corda-updates"])
+    private var configFile : String? = null
+
+    override fun runProgram() : Int {
+        return when (command) {
+            CLICommands.INIT -> init()
+            CLICommands.SYNC -> sync()
+            CLICommands.PRINT_VERSIONS -> printVersions()
+        }
+    }
+
+    private fun init() : Int {
+        logger.info("Launching in ${CLICommands.INIT} mode")
+
+        val basePath = if (configFile == null) File(System.getProperty("user.home"), ".corda-updates") else File(configFile)
+
+        logger.info("Initialising in ${basePath.canonicalPath}")
+        if (basePath.exists()) {
+            print("Path ${basePath.canonicalPath} already exists. Please remove the folder before initialising, to avoid any of its contents from being overwritten.")
+            return ExitCodes.FAILURE
+        }
+
+        val repoPath = File(basePath, "repo")
+
+        Files.createDirectories(basePath.toPath())
+        Files.createDirectories(repoPath.toPath())
+
+        // creating default settings file from a template
+        val mf = DefaultMustacheFactory()
```

Comment:
Do we need a full blown template engine for this? Is it so complex? Can a simple string interpolation work just as well?

Reply:
Moved corda-updates-shell to a separate branch.